src/main/java/org/jitsi/videobridge/BitrateController.java

```diff
@@ -0,0 +1,601 @@
+
+                    if (rtpEncodings[j].getRTXSSRC() != -1)
+                    {
+                        subCtrls.put((int) rtpEncodings[j].getRTXSSRC(), ctrl);
+                    }
+                }
+            }
+
+            if (ctrl != null)
+            {
+                ctrl.update(targetIdx);
+            }
+
+            if (targetIdx > -1)
+            {
+                newForwardedEndpoints.add(allocations[i].endpointID);
+            }
+        }
+
+        this.forwardedEndpoints = newForwardedEndpoints;
+    }
+
+    /**
+     * Gets the {@link List} of endpoints that are currently being forwarded,
+     * represented by their IDs.
+     *
+     * @return the {@link List} of endpoints that are currently being forwarded,
+     * represented by their IDs.
+     */
+    public List<String> getForwardedEndpoints()
+    {
+        return forwardedEndpoints;
+    }
+
+    /**
+     * Checks whether RTP packets from {@code source} should be forwarded
+     * to {@link #dest}.
+     * @param source the channel.
+     * @return {@code true} iff RTP packets from {@code source} should
+     * be forwarded to {@link #dest}.
+     */
+    public boolean isForwarded(RtpChannel source)
+    {
+        String endpointID = source.getEndpoint().getID();
+        return forwardedEndpoints.contains(endpointID);
+    }
+
+    /**
+     * The {@link PacketTransformer} that handles incoming/outgoing RTP
+     * packets for this {@link BitrateController} instance. Internally,
+     * it delegates this responsibility to the appropriate media stream track
+     * bitrate controller ({@link SimulcastController}, etc).
+     */
+    class RTPTransformer
+        implements PacketTransformer
+    {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void close()
+        {
+            // TODO decrease counters.
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] reverseTransform(RawPacket[] pkts)
+        {
+            return pkts;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] transform(RawPacket[] pkts)
+        {
+            if (ArrayUtils.isNullOrEmpty(pkts))
+            {
+                return pkts;
+            }
+
+            for (int i = 0; i < pkts.length; i++)
+            {
+                if (pkts[i] == null
+                    || !RTPPacketPredicate.INSTANCE.test(pkts[i]))
+                {
+                    continue;
+                }
+
+                int ssrc = pkts[i].getSSRC();
+                SimulcastController subCtrl = subCtrls.get(ssrc);
+
+                if (subCtrl == null)
+                {
+                    pkts[i] = null;
+                    continue;
+                }
+
+                RawPacket[] transformedPkts = subCtrl
+                    .getRTPTransformer().transform(new RawPacket[]{pkts[i]});
```

Comment:
If we use the SinglePacketTransformer here (and in SimulcastController) it will be simpler and we will not unnecessarily create a new array instance on every packet.